active/0000-blueprint-remove-old-files.md

```diff
@@ -0,0 +1,48 @@
+- Start Date: 2016-12-17
+- RFC PR: (leave this empty)
+- Ember CLI Issue: (leave this empty)
+
+# Summary
+
+Give blueprint generators the ability to clean up old files.
+
+# Motivation
+
+We want to eliminate the noise of having old files laying after updating
+ember-cli using `ember init`.
+
+# Detailed design
+
+We'd like a declarative API for blueprints to delete files instead of only
+create. It would be essentially syntactic sugar for removing the file yourself
+in an `afterInstall` hook. It would be an array on the blueprint's `index.js`.
```

Comment:
I was thinking it would be as basic as possible, and let git handle anything more complicated.
* eslint removes .jshintrc
* git commit
* some time passes
* ember init adds it back
* git tells you this file you don't want was added back, so you manually remove the file

If we started composing, I think there would be many corner cases with a lot of complexity. My goal is to get old files out of the project that until now were invisible to you. With my steps above, it is now visible to you that it was removed for some reason, even though there are some extra steps to get rid of it again.